functions-and-scope.js

```diff
@@ -58,26 +132,56 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
+/* Bonusopdracht: hoogste cijfer */
 
 
 
-/* Bonusopdracht: hoogste cijfer */
-
 /* 3a: Script schrijven  */
 // Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
-// * Hoe kan ik iedere waarde van de array langsgaan?
-// * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
-// Log het antwoord in de terminal.
 
-// ---- Verwachte uitkomst: 9
+//---------------------stappenplan--------------------------//
+
+// Stap 1 Maak een functie die highestGrade heet.
+// Stap 2 Maak een variabele die het hoogste cijfer opslaat
+// Stap 2 Zoek het hoogste cijfer in de lijst op (for loop)
+
+//----------------------------------------------------------//
+
+function highestGrade(highestGrade) {
+    let highGrade = 0;
```

Comment:
Het is altijd beter om te beginnen met de eerste waarde van de array. Beetje flauw bij dit voorbeeld, maar stel dat je een array hebt van `[-2, -5, -1, -3]`. Als nu deze array aan jouw functie mee zou geven, dan komt daar `0` uit, terwijl dat cijfer helemaal niet in de array voorkomt.

```suggestion
    let highGrade = highestGrade[0];
```